nemo/collections/tts/models/base.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+
+from nemo.core.classes import ModelPT
+
+
+class SpectrogramGenerator(ModelPT, ABC):
+    """ Base class for all TTS models that turn text into a spectrogram """
+
+    @abstractmethod
+    def parse(self, str_input: str, **kwargs) -> 'torch.tensor':
```

Comment:
What if how models choose to parse their text differs from model to model?

Reply:
Is it bad practice to add **kwargs for abstractmethods?

Reply:
Sunclasses can override parent signatures, with same function name. (python advices against this, cause method overriding doesn't exist, but technically there is no issue since it will dispatch to the sunclasses function pointer in the stack). For an example, look at how change_vocabulary() works in the base CTC model and it's subclass in ASR BPE models. 

Note, doing this means auto complete will kinda fail, so that's there.